File: Solution3.py
class Solution(object):
    def findPeakElement(self, nums):
        low, high = 0, len(nums) - 1
        while low <= high:
            mid = low + (high - low)/2
            if (mid==0 or nums[mid] >= nums[mid-1]) and (mid == len(nums) - 1 or nums[mid] >= nums[mid+1]):
                return mid
            elif mid + 1 and nums[mid] < nums[mid+1]:
                low = mid + 1
            else:
                high = mid - 1
        return -1

File: Solution3.java
/*
    Time Complexity : O(logn)
    Space Complexity : O(1)
    Accepted LeetCode URL : https://leetcode.com/submissions/detail/633802508/
*/
class Solution {
    public int findPeakElement(int[] nums) {
        int lo = 0;
        int hi = nums.length - 1;
        
        while(lo < hi){
            int mid = lo + (hi - lo)/2;
            
            if(nums[mid] < nums[mid + 1]){
                lo = mid + 1;
            }else{
                hi = mid;
            }
        }
        
        return lo;
    }
}
